app/authentication/jti_claim_storage.py

```diff
@@ -0,0 +1,39 @@
+from uuid import UUID
+
+from sqlalchemy.exc import IntegrityError
+from structlog import get_logger
+
+from app.data_model.database import UsedJtiClaim, commit_or_rollback, db_session
+
+logger = get_logger()
+
+
+def _is_valid(jti_claim):
+    try:
+        UUID(jti_claim, version=4)
+    except ValueError:
+        return False
+    return True
+
+
+def use_jti_claim(jti_claim):
+    """
+    Use a jti claim if it is supplied
+    :param jti_claim: jti claim to mark as used. Must be a valid uuid 4.
+    :return: True if the token was used or wasn't supplied and False if the token is invalid or has already been used.
+    """
+    if jti_claim is None:
+        return True
```

Comment:
Was how I initially implemented and I also think it's not as clear. Can change back if @collisdigital is happy with that